packages/nextjs/src/sharedTypes/component-props.ts

```diff
@@ -21,27 +21,20 @@ export type ComponentPropsFetchFunction<NextContext, FetchedProps = unknown> = {
 };
 
 /**
- * Shape of getServerSideProps function on component level
+ * Common server component props used in both SSG and SSR modes
  */
-export type GetServerSideComponentProps = ComponentPropsFetchFunction<GetServerSidePropsContext>;
-
-/**
- * Shape of getStaticProps function on component level
- */
-export type GetStaticComponentProps = ComponentPropsFetchFunction<GetStaticPropsContext>;
+export type GetComponentServerProps = ComponentPropsFetchFunction<
```

Comment:
See my explanation regarding "server" part in naming
```suggestion
export type GetComponentProps = ComponentPropsFetchFunction<
```

Reply:
let's have this convo in the thread above